scripts/dtsBundler.mjs

```diff
@@ -310,6 +318,10 @@ function verifyMatchingSymbols(decl) {
             }
             const symbolInScope = findInScope(symbolOfNode.name);
             if (!symbolInScope) {
+                if (symbolOfNode.declarations?.every(d => isLocalDeclaration(d) && d.getSourceFile() === decl.getSourceFile()) && !isSelfReference(node, symbolOfNode)) {
+                    // The symbol is a local that needs to be copied into the scope.
+                    scopeStack[scopeStack.length - 1].locals.set(symbolOfNode.name, { symbol: symbolOfNode, writeTarget: isInternal ? WriteTarget.Internal : WriteTarget.Both });
+                }
                 // We didn't find the symbol in scope at all. Just allow it and we'll fail at test time.
                 return node;

```

Comment:
Maybe not for this PR, but I do feel a bit like I should have originally just made this error.